### PR TITLE
Add generated Icon diagram to api docs

### PR DESCRIPTION
**Pull request by brianegan**
## Description

This PR embeds a new Icon diagram to match the sample code that is in the api docs.

## Related Issues

  - [x] Before merge, the asset must be merged: https://github.com/flutter/assets-for-api-docs/pull/88

## Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [X] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/icon.dart [hunk 1]
```diff
@@ -29,6 +29,8 @@ import 'icon_theme_data.dart';
 /// This example shows how to use [Icon] to create an addition icon, in the
 /// color pink, and 30 x 30 pixels in size.
 ///
+/// ![A pink plus sign](https://raw.githubusercontent.com/flutter/assets-for-api-docs/master/assets/widgets/icon.png)
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/icon.dart]:
```suggestion
/// ![A pink plus sign](https://flutter.github.io/flutter/assets-for-api-docs/master/assets/widgets/icon.png)
```

**Review by Piinks — COMMENTED** [review]
Once https://github.com/flutter/assets-for-api-docs/pull/88 lands, this link should resolve properly. :+1:

File: packages/flutter/lib/src/widgets/icon.dart [hunk 2]
```diff
@@ -29,6 +29,8 @@ import 'icon_theme_data.dart';
 /// This example shows how to use [Icon] to create an addition icon, in the
 /// color pink, and 30 x 30 pixels in size.
 ///
+/// ![A pink plus sign](https://flutter.github.io/flutter/assets-for-api-docs/master/assets/widgets/icon.png)
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/icon.dart]:
```suggestion
/// ![A pink plus sign](https://flutter.github.io/assets-for-api-docs/assets/widgets/icon.png)
```
ack sorry. I didn't notice the other segments in the url. This one works. :)

Comment (brianegan) [reply packages/flutter/lib/src/widgets/icon.dart]: Thanks!

File: packages/flutter/lib/src/widgets/icon.dart [hunk 3]
```diff
@@ -133,7 +135,8 @@ class Icon extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     assert(this.textDirection != null || debugCheckHasDirectionality(context));
-    final TextDirection textDirection = this.textDirection ?? Directionality.of(context);
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/icon.dart]: Nit: No need to break the line

File: packages/flutter/lib/src/widgets/icon.dart [hunk 4]
```diff
@@ -153,7 +156,8 @@ class Icon extends StatelessWidget {
 
     Widget iconWidget = RichText(
       overflow: TextOverflow.visible, // Never clip.
-      textDirection: textDirection, // Since we already fetched it for the assert...
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/icon.dart]: same formatting comment for here and below.

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

**Pull request merged**